### PR TITLE
Quiz

### PR DESCRIPTION
**Pull request by SkyDiverStar**
# Description of changes

grafik
grafik
grafik
grafik

## How has this been tested?

Please describe the test strategy you followed.

- [ ] automated unit test
- [ ] automated integration test
- [ ] automated acceptance test
- [x] manual, exploratory test

In case of manual test, please document the test well including a set of user instructions and prerequisites. Each including an action, it's result, and where appropriate a screenshot.

## Checklist before requesting a review

- [ ] My code follows the [coding guidelines](https://github.com/IT-REX-Platform/wiki/blob/main/dev-manuals/backend/coding-guidelines.md) of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] My code fulfilles all acceptance criteria
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have made corresponding changes to the documentation
- [ ] I have added explanation of architectural decision and rationales to [wiki/adr](https://github.com/IT-REX-Platform/wiki/tree/main/adr)
- [ ] I have updated the changes in the ticket description

## Checklist for reviewer

- [ ] The code works and does not throw errors
- [ ] The code is easy to understand and there are no confusing parts
- [ ] The code follows the [coding guidelines](https://github.com/IT-REX-Platform/wiki/blob/main/dev-manuals/backend/coding-guidelines.md) of this project
- [ ] The code change accomplishes what it is supposed to do
- [ ] I cannot think of any use case in which the code does not behave as intended
- [ ] The added and existing tests reasonably cover the code change
- [ ] I cannot think of any test cases, input or edge cases that should be tested in addition
- [ ] Description of the change is included in the documentation

### CODE REVIEW

File: app/courses/[courseId]/quiz/[quizId]/student.tsx [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+              }
+            }
+          }
+        }
+      }
+    `,
+    { id: [quizId] }
+  );
+
+  const quiz = contentsByIds[0].quiz;
```
Comment (mardom1) [inline app/courses/[courseId]/quiz/[quizId]/student.tsx:93]: You should check if the content exists and return a 404 page if not.

File: app/courses/[courseId]/quiz/[quizId]/student.tsx [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+        </div>
+      </div>
+
+      <div className="flex justify-center gap-4">
+        <FormGroup>
+          {answers!.map((answer, index) => (
+            <div key={index}>
+              <FormControlLabel
+                control={
+                  <Checkbox
```
Comment (mardom1) [inline app/courses/[courseId]/quiz/[quizId]/student.tsx:151]: As they are single-choice, radio-buttons would be a better. If they should be multiple choice, I can only select one

File: app/courses/[courseId]/quiz/[quizId]/student.tsx [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+                    disabled={checkAnswers}
+                    style={{
+                      color:
+                        checkAnswers && answer.correct
```
Comment (mardom1) [inline app/courses/[courseId]/quiz/[quizId]/student.tsx:155]: I would make them gray if not selected

File: app/courses/[courseId]/quiz/[quizId]/student.tsx [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+        <div className="bg-white -mb-[9px] px-3 text-xs text-gray-600">
+          {currentIndex + 1}/{quiz?.selectedQuestions.length ?? 0}
+        </div>
+      </div>
+
+      <div className="mt-6 text-center text-gray-600">{questionText}</div>
+
+      <div className="w-full border-b border-b-gray-300 mt-6 flex justify-center mb-6">
+        <div>
+          <IconButton onClick={() => setInfoDialogOpen(true)}>
```
Comment (mardom1) [inline app/courses/[courseId]/quiz/[quizId]/student.tsx:139]: Not sure, but a text button with "Hints" would be clearer here? Or maybe a tooltip

**Review by mardom1 — CHANGES_REQUESTED** [review]

File: app/courses/[courseId]/quiz/[quizId]/page.tsx [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+import { PageView, usePageView } from "@/src/currentView";
+import StudentQuiz from "./student";
+//import EditQuiz from "./lecturer";
+
+export default function CoursePage() {
+  const [pageView, _] = usePageView();
+  switch (pageView) {
+    case PageView.Student:
+      return <StudentQuiz />;
+    //case PageView.Lecturer:
```
Comment (mardom1) [inline app/courses/[courseId]/quiz/[quizId]/page.tsx:12]: I would return a 404 for now

File: app/courses/[courseId]/quiz/[quizId]/student.tsx (hunk 2 above)
Comment (SkyDiverStar) [reply app/courses/[courseId]/quiz/[quizId]/student.tsx:151]: But they are not only single choice. In some quizzes you can have more than 1 answer.

File: app/courses/[courseId]/quiz/[quizId]/student.tsx (hunk 4 above)
Comment (SkyDiverStar) [reply app/courses/[courseId]/quiz/[quizId]/student.tsx:139]: Done as button

File: app/courses/[courseId]/quiz/[quizId]/student.tsx [hunk 6]
```diff
@@ -0,0 +1,231 @@
...
+  const [checkAnswers, setCheckAnswers] = useState(false);
+  const [userAnswers, setUserAnswers] = useState<UserAnswers>([]);
+
+  useEffect(() => {
+    // Reset user answers when the currentIndex changes
+    setUserAnswers([]);
+  }, [currentIndex]);
+
+  // Fetch quiz data
+  const { contentsByIds, currentUserInfo } = useLazyLoadQuery<studentQuizQuery>(
```
Comment (mardom1) [inline app/courses/[courseId]/quiz/[quizId]/student.tsx:40]: `currentUserInfo` is unused

**Review by mardom1 — APPROVED** [review]

File: app/courses/[courseId]/quiz/[quizId]/student.tsx (hunk 6 above)
Comment (SkyDiverStar) [reply app/courses/[courseId]/quiz/[quizId]/student.tsx]: Removed
